proto/desmos/reports/v1/query.proto

```diff
@@ -0,0 +1,86 @@
+syntax = "proto3";
+package desmos.reports.v1;
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/any.proto";
+import "cosmos_proto/cosmos.proto";
+import "cosmos/base/query/v1beta1/pagination.proto";
+
+import "desmos/reports/v1/models.proto";
+
+option go_package = "github.com/desmos-labs/desmos/v3/x/reports/types";
+
+// Query defines the gRPC querier service.
+service Query {
+  // Reports allows to query the reports for a specific target
+  rpc Reports(QueryReportsRequest) returns (QueryReportsResponse) {
+    option (google.api.http).get = "/desmos/reports/v1/{subspace_id}/reports";
+  }
+
+  // Reasons allows to query the supported reporting reasons for a subspace
+  rpc Reasons(QueryReasonsRequest) returns (QueryReasonsResponse) {
+    option (google.api.http).get = "/desmos/reports/v1/{subspace_id}/reasons";
+  }
+
+  // Params allows to query the module parameters
+  rpc Params(QueryParamsRequest) returns (QueryParamsResponse) {
+    option (google.api.http).get = "/desmos/reports/v1/params";
+  }
+}
+
+// QueryReportsResponse is the request type for Query/Reports RPC method
+message QueryReportsRequest {
+  // Id of the subspace to query the reports for
+  uint64 subspace_id = 1 [ (gogoproto.moretags) = "yaml:\"subspace_id\"" ];
+
+  // (optional) Target to query the reports for
+  google.protobuf.Any target = 2 [ (gogoproto.moretags) = "yaml:\"target\"" ];
+
```

Comment:
Since we have the content key showing the relationship between the reported content and reporter, adding a `reporter` field would be useful to get the user report on a content (a post or a user).

Reply:
Makes sense. I've now added this ability to the `Reports` query: the `reporter` field will be used only if a `target` is specified.